tests/rust/pipe-test/create_pipe/mod.rs

```diff
@@ -16,32 +16,13 @@ use ::demikernel::{
 //======================================================================================================================
 
 /// Drives integration tests for pipe queues.
-pub fn run(libos: &mut LibOS, pipe_name: &str) -> Result<()> {
-    let mut ret: Result<(), anyhow::Error> = Ok(());
+pub fn run(libos: &mut LibOS, pipe_name: &str) -> usize {
+    let mut nfailed: usize = 0;
 
-    match create_pipe_with_invalid_name(libos) {
-        Ok(_) => println!("[passed] {}", stringify!(create_pipe_with_invalid_name)),
-        Err(e) => {
-            // Don't overwrite previous error.
-            if ret.is_ok() {
-                ret = Err(e);
-            }
-            println!("[FAILED] {}", stringify!(create_pipe_with_invalid_name))
-        },
-    };
-
-    match create_pipe_with_same_name(libos, pipe_name) {
-        Ok(_) => println!("[passed] {}", stringify!(create_pipe_with_same_name)),
-        Err(e) => {
-            // Don't overwrite previous error.
-            if ret.is_ok() {
-                ret = Err(e);
-            }
-            println!("[FAILED] {}", stringify!(create_pipe_with_same_name))
-        },
-    };
+    nfailed += crate::test!(create_pipe_with_invalid_name(libos));
+    nfailed += crate::test!(create_pipe_with_same_name(libos, pipe_name));
```

Comment:
Rather than just storing the first error and then incrementing a counter on each test, perhaps we should collect all of the errors/results into an Vec? If we don't print them, then it probably doesn't matter whether we keep the first or last one (or even just keep a flag). If we do print them, then we should probably print all of them, regardless of whether we encountered an error before.

Reply:
I agree that collecting results would be the best solution possible - we are currently printing all failures, the end result of each test, and incrementing the failed counter to produce final report.

When you suggest to collect all errors/results into a vector, do you mean that we should change the return value of each test function to `Result<(), Vector::<anyhow::Error>>`?

Reply:
No, I recommend that crate::test! still return a result and nfailed be a vector instead of a usize. Then we can choose to print all the errors at the end and/or count them.

Reply:
I see. But then we wouldn't be actually collecting all the failures that happened in the test, right?

Reply:
Cool, base don our offline discussion, I've introduced a new macro that collects results in a vector.